### PR TITLE
Add stuff for image search

### PR DESCRIPTION
**Pull request by hwkr**
- Set up timeout for input field
- Improve error handling and fetch call

### CODE REVIEW

**Comment by hwkr** [comment]
Two main changes,
- Tweaked the colors in the header

image

- Changed how the image field works including adding error states

image image

**Comment by hwkr** [comment]
- Added a back button to sign up for @Inschato 

image image

File: client/components/parts/HeroImageField.jsx [hunk 1]
```diff
@@ -1,82 +1,66 @@
...
 import ImageUploadField from 'parts/ImageUploadField'
-import classNames from 'classnames'
 
-export default class HeroImageField extends Component{
+export default class HeroImageField extends PureComponent{
   constructor(props){
     super(props)
     this.state = {
-      tabStates: this.getInitialTabStates()
+      activeTab: 0
```
Comment (bndnio) [inline client/components/parts/HeroImageField.jsx:13]: how to we map tab enumeration to the types?

File: client/components/parts/ImageSearchField.jsx [hunk 2]
```diff
@@ -28,63 +27,64 @@ export default class ImageSearchField extends Component {
   }
 
   handleChange = (e) => {
-    this.props.updateImage(e.target.id)
+    this.props.updateImage(e.target.dataset['url'])
```
Comment (bndnio) [inline client/components/parts/ImageSearchField.jsx]: why not `dataset.url`? 
shouldn't `[<varname>]` be used only in the scenario where `varname` is dynamic?

File: client/components/parts/ImageSearchField.jsx [hunk 3]
```diff
@@ -28,63 +27,64 @@ export default class ImageSearchField extends Component {
...
+              'full': imgObject.urls.full
+            })
+          )))
+        } else {
+          reject(response.status)
+        }
       })
-    }))
-  }
+    }, 500) // <- Wait 500ms before querying
```
Comment (bndnio) [inline client/components/parts/ImageSearchField.jsx:54]: Yay timeouts! 
The global this.searchTimer hurts my heart a bit but it's worth it

File: client/components/parts/ImageSearchField.jsx [hunk 4]
```diff
@@ -28,63 +27,64 @@ export default class ImageSearchField extends Component {
...
+            then={(imgUrls) =>
+              <div className="columns col-gapless">
+                {imgUrls.slice(0, 4).map((urls, index) =>
+                  <div className="column col-6 thumbnail" key={`thumbnail-${index}`}>
+                    <button type="button" className="btn btn-link btn-thumbnail" onClick={this.handleChange}>
+                      <img className="img-responsive rounded" src={urls.thumbnail} data-url={urls.full} />
+                    </button>
+                  </div>
+                  )}
+              </div>
```
Comment (bndnio) [inline client/components/parts/ImageSearchField.jsx:73]: I think in the future it'd be better practice to write functions that output html above the return statement in the `render()` function. This is fine for now though

File: client/components/parts/ImageSearchField.jsx [hunk 5]
```diff
@@ -28,63 +27,64 @@ export default class ImageSearchField extends Component {
...
+  // Clubhub Client ID: d9ac3268ec6e896a2f2655d25b135b0d22dab5762719fbe4fbc25864860ab8e0
+  // Juan's  Client ID: ef9e27a50e4d36d08c24bb35f362974c36420854e6924ad0e4b79c4c6dd8b041
+
+  searchByKeyword = () => new Promise((resolve, reject) => {
+    clearTimeout(this.searchTimer)
+    this.searchTimer = setTimeout(() => {
+      // Send search request
+      fetch(`https://api.unsplash.com/search/photos/?client_id=${Config.unsplash_client_id}&query=${this.state.searchKeywords}`)
+      .then((response) => {
+        if (response.ok){
```
Comment (bndnio) [inline client/components/parts/ImageSearchField.jsx:43]: I like the response error checking 👍

File: config.dev.json [hunk 6]
```diff
@@ -7,5 +7,6 @@
     "cookie_path": [
         ".uvic.club.localhost"
     ],
+    "unsplash_client_id": "ef9e27a50e4d36d08c24bb35f362974c36420854e6924ad0e4b79c4c6dd8b041",
```
Comment (bndnio) [inline config.dev.json:10]: nice work on moving things to config 👍

**Review by bndnio — COMMENTED** [review]
Looks good!
Most comments are questions or forward looking suggestions (i.e. for further work/PRs)

File: client/components/parts/HeroImageField.jsx (hunk 1 above)
Comment (hwkr) [reply client/components/parts/HeroImageField.jsx:13]: What? I don't understand the question

File: client/components/parts/ImageSearchField.jsx (hunk 2 above)
Comment (hwkr) [reply client/components/parts/ImageSearchField.jsx]: Sure, it's just syntax

File: client/components/parts/ImageSearchField.jsx (hunk 4 above)
Comment (hwkr) [reply client/components/parts/ImageSearchField.jsx:73]: I actually prefer it in the HTML, it's more like templating which I think is easier to understand for most people.

File: client/components/parts/HeroImageField.jsx (hunk 1 above)
Comment (bndnio) [reply client/components/parts/HeroImageField.jsx:13]: tabSates used to be an array of strings with true/false correct? (or at least something similar).
Now active tab is a number - how does each number map to a tab name?

File: client/components/parts/ImageSearchField.jsx (hunk 4 above)
Comment (bndnio) [reply client/components/parts/ImageSearchField.jsx:73]: I think the issue is more with the nested functions. e.g. having an `<ASync>` component with a `then` prop which is an inline function with a `.map`. 
I'm fine keeping it either way, just so long as everyone's on the same page

**Comment by bndnio** [comment]
Looks great! You're clear to merge on my end 👍

**Pull request merged**